node/modules/chain.go

```diff
@@ -175,6 +231,68 @@ func NewSyncer(params SyncerParams) (*chain.Syncer, error) {
 	return syncer, nil
 }
 
+func NewSyncerFollower(params SyncerFollowerParams) (*chain.Syncer, error) {
+	var (
+		g  = params.GatewayAPI
+		lc = params.Lifecycle
+		sm = params.StateManager
+	)
+	lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
```

Comment:
The OnStart context is cancelled after the startup sequence, so you really don't want to use that one.

The correct way to get a context in constructors is with `ctx := helpers.LifececleCtx(mctx, lc)`, where mctx is iirc `dtypes.MetrictCtx` (metricsCtx is a special root context that never gets cancelled, but carries values used for collecting opencensus metrics/tracing, LifececleCtx connects it to a lifecycle which ensures correct order for stopping components - so the context you get from there will be cancelled after everything else that depends on the syncer has stopped)